packages/trader/src/Modules/Trading/Containers/trade.tsx

```diff
@@ -0,0 +1,235 @@
+        is_dark_mode_on: is_dark_theme,
+        is_mobile,
+        notification_messages_ui: NotificationMessages,
+        should_show_multipliers_onboarding,
+    } = ui;
+    const { is_eu } = client;
+    const { network_status } = common;
+
+    const [digits, setDigits] = React.useState<number[]>([]);
+    const [tick, setTick] = React.useState<null | TickSpotData>(null);
+    const [try_synthetic_indices, setTrySyntheticIndices] = React.useState(false);
+    const [try_open_markets, setTryOpenMarkets] = React.useState(false);
+    const [category, setCategory] = React.useState<string>();
+    const [subcategory, setSubcategory] = React.useState<string>();
+    const [swipe_index, setSwipeIndex] = React.useState<number | undefined>(0);
+
+    const open_market = React.useMemo(() => {
+        if (try_synthetic_indices) {
+            return { category: 'synthetics' };
+        } else if (try_open_markets && category) {
+            return { category, subcategory };
+        }
+        return null;
+    }, [try_synthetic_indices, try_open_markets, category, subcategory]);
+
+    React.useEffect(() => {
+        onMount();
+        if (!is_synthetics_available) {
+            const setMarket = async () => {
+                const markets_to_search = ['forex', 'indices', 'commodities']; // none-synthetic
+                const { category: market_cat, subcategory: market_subcat } =
+                    (await getFirstOpenMarket(markets_to_search)) ?? {};
+                if (market_cat) {
+                    setCategory(market_cat);
+                    setSubcategory(market_subcat);
+                }
+            };
+
+            setMarket();
+        }
+        return () => onUnmount();
+    }, [onMount, onUnmount, getFirstOpenMarket, is_synthetics_available]);
+
+    React.useEffect(() => {
+        if (is_mobile) {
+            setDigits([]);
+        }
+        setTrySyntheticIndices(false);
+        setTryOpenMarkets(false);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [symbol, setDigits, setTrySyntheticIndices, is_synthetics_available]);
+
+    React.useEffect(() => {
+        const selectMultipliers = async () => {
+            await setContractTypes();
+
+            onChange({ target: { name: 'contract_type', value: 'multiplier' } });
+        };
+        if (should_show_multipliers_onboarding && !is_chart_loading && (is_synthetics_available || !is_market_closed)) {
+            selectMultipliers();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [should_show_multipliers_onboarding, is_chart_loading]);
+
+    const bottomWidgets = React.useCallback(({ digits: d, tick: t }: TBottomWidgetsParams) => {
+        return <BottomWidgetsMobile digits={d} tick={t} setTick={setTick} setDigits={setDigits} />;
+    }, []);
+
+    const onChangeSwipeableIndex = (index?: number) => {
+        setMobileDigitView(index === 0);
```

Comment:
Is TS not complaining that we are comparing value, which might be `undefined` with `0` (number)? I mean without `Number(index)`

Reply:
@kate-deriv no, it's not complaining :)